### PR TITLE
Adding fields for MediaStreams and children

### PR DESCRIPTION
**Pull request by shearn89**
The existing library is missing the `MediaStreams` object as generated by `sam`:

```
❯ sam local generate-event connect contact-flow-event
{
  "Name": "ContactFlowEvent",
  "Details": {
    "ContactData": {
      "Attributes": {},
      "Channel": "VOICE",
      "ContactId": "5ca32fbd-8f92-46af-92a5-6b0f970f0efe",
      "CustomerEndpoint": {
        "Address": "+11234567890",
        "Type": "TELEPHONE_NUMBER"
      },
      "InitialContactId": "5ca32fbd-8f92-46af-92a5-6b0f970f0efe",
      "InitiationMethod": "API",
      "InstanceARN": "arn:aws:connect:us-east-1:123456789012:instance/9308c2a1-9bc6-4cea-8290-6c0b4a6d38fa",
      "MediaStreams": {
        "Customer": {
          "Audio": {
            "StartFragmentNumber": "91343852333181432392682062622220590765191907586",
            "StartTimestamp": "1565781909613",
            "StreamARN": "arn:aws:kinesisvideo:us-east-1:123456789012:stream/connect-contact-a3d73b84-ce0e-479a-a9dc-5637c9d30ac9/1565272947806"
          }
        }
      },
      "PreviousContactId": "5ca32fbd-8f92-46af-92a5-6b0f970f0efe",
      "Queue": null,
      "SystemEndpoint": {
        "Address": "+11234567890",
        "Type": "TELEPHONE_NUMBER"
      }
    },
    "Parameters": {}
  }
}
```

*Description of changes:*

Add in fields for `MediaStreams` and below, including `StopFragmentNumber` which appears to be undocumented but exists based on my testing.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/ConnectEvent.java [hunk 1]
```diff
@@ -72,6 +72,33 @@ public static class CustomerEndpoint implements Serializable, Cloneable {
         private String type;
     }
 
+        @Data
```
Comment (andclt) [inline aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/ConnectEvent.java:75]: nit: indentation

File: aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/ConnectEvent.java [hunk 2]
```diff
@@ -72,6 +72,33 @@ public static class CustomerEndpoint implements Serializable, Cloneable {
...
+    @Builder(setterPrefix = "with")
+    @NoArgsConstructor
+    @AllArgsConstructor
+    public static class MediaStreams implements Serializable {
```
Comment (andclt) [inline aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/ConnectEvent.java:79]: Looking at the event you pasted in the PR's description, you should add `MediaStreams` class to the `ContactData` class

**Comment by andclt** [comment]
Could you please set the destination branch to `eventsv4-serializationv2`?
Furthermore, it would be nice if you could update the event model `ConnectEvent.json` in the test folder.

**Comment by shearn89** [comment]
Thanks @andclt - I believe I've addressed all your comments, if you can take another look through and let me know if further changes are needed!

**Comment by andclt** [comment]
Thank you very much @shearn89! 
I'm sorry I forgot to say it before but could you also update this JSON file here please https://github.com/aws/aws-lambda-java-libs/blob/events-v4-serialization-v2/aws-lambda-java-events/src/test/resources/event_models/connect_event.json?

That should be the last change required :)

**Comment by shearn89** [comment]
Ah perfect - I'd updated the wrong event, hopefully fixed now 🤞🏻 !

**Comment by andclt** [comment]
The first job failed because of this recent change: https://github.com/aws/aws-lambda-java-libs/pull/376 

Adding `aws-lambda-java-core` here https://github.com/aws/aws-lambda-java-libs/blob/events-v4-serialization-v2/aws-lambda-java-tests/pom.xml should fix the issue.
```
 <dependency>
      <groupId>com.amazonaws</groupId>
      <artifactId>aws-lambda-java-core</artifactId>
      <version>1.2.2</version>
  </dependency>
```

**Comment by andclt** [comment]
LGTM, thanks again @shearn89

**Review by andclt — APPROVED** [review]

**Comment by msailes** [comment]
Is this documented anywhere else other than SAM generate-event?

**Comment by andclt** [comment]
> Is this documented anywhere else other than SAM generate-event?

I found this in the public docs: https://docs.aws.amazon.com/connect/latest/adminguide/media-streaming-attributes.html

**Comment by msailes** [comment]
Could that be added somewhere in the Javadocs, so we can keep track of all the features.

**Comment by shearn89** [comment]
Sorry, this dropped off my radar for a bit whilst busy. I've added a comment to the `MediaStreams` class to keep the note as close to the object as I can.

**Comment by andclt** [comment]
LGTM, thanks again!

**Pull request merged**